# Importing important libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
ies_df = pd.read_csv('all_energy_statistics.csv')
ies_df.shape

(1189482, 7)

dataset has 1189482 rows and 7 columns

# Exploratory Data Analysis

In [3]:
ies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1189482 entries, 0 to 1189481
Data columns (total 7 columns):
 #   Column                 Non-Null Count    Dtype  
---  ------                 --------------    -----  
 0   country_or_area        1189482 non-null  object 
 1   commodity_transaction  1189482 non-null  object 
 2   year                   1189482 non-null  int64  
 3   unit                   1189482 non-null  object 
 4   quantity               1189482 non-null  float64
 5   quantity_footnotes     163946 non-null   float64
 6   category               1189482 non-null  object 
dtypes: float64(2), int64(1), object(4)
memory usage: 63.5+ MB


In [4]:
ies_df.head()

country_or_area               commodity_transaction  year  \
0         Austria  Additives and Oxygenates - Exports  1996   
1         Austria  Additives and Oxygenates - Exports  1995   
2         Belgium  Additives and Oxygenates - Exports  2014   
3         Belgium  Additives and Oxygenates - Exports  2013   
4         Belgium  Additives and Oxygenates - Exports  2012   

                     unit  quantity  quantity_footnotes  \
0  Metric tons,  thousand       5.0                 NaN   
1  Metric tons,  thousand      17.0                 NaN   
2  Metric tons,  thousand       0.0                 NaN   
3  Metric tons,  thousand       0.0                 NaN   
4  Metric tons,  thousand      35.0                 NaN   

                   category  
0  additives_and_oxygenates  
1  additives_and_oxygenates  
2  additives_and_oxygenates  
3  additives_and_oxygenates  
4  additives_and_oxygenates

In [5]:
def print_unique(df):
    '''
    This method takes in a pandas dataframe,
    and prints out the unique values for each column
    '''
    for col in df.columns:
        print(col + " : " + str(df[col].nunique()))

In [6]:
#let us see the number of unique items in each column
print_unique(ies_df)

country_or_area : 243
commodity_transaction : 2452
year : 25
unit : 6
quantity : 137659
quantity_footnotes : 1
category : 71


In [7]:
ies_df['category'].value_counts().index[0]

'total_electricity'

In [8]:
#dropping footnotes column
ies_df.drop(columns=['quantity_footnotes'], inplace=True)
ies_df.shape

(1189482, 6)

In [9]:
com_df = ies_df.groupby('commodity_transaction')

In [10]:
len(com_df)

2452

Electricity - total wind production, Electricity - total solar production

In [11]:
electricity_set = set()

for i in range (0, len(com_df)):
    for entry in com_df['commodity_transaction'].value_counts().index[i]:
        if str(entry).startswith('Electricity'):
            electricity_set.add(entry)

len(electricity_set)

KeyboardInterrupt: 